samples/sampler_filter.html

```diff
@@ -0,0 +1,302 @@
+        #define TEXCOORD 4
+
+        precision highp float;
+        precision highp int;
+
+        uniform mat4 mvp;
+
+        layout(location = POSITION) in vec2 position;
+        layout(location = TEXCOORD) in vec2 textureCoordinates;
+
+        out vec2 st;
+
+        void main()
+        {
+            st = textureCoordinates;
+            gl_Position = mvp * vec4(position, 0.0, 1.0) ;
+        }
+    </script>
+
+    <script id="fs" type="x-shader/x-fragment">
+        #version 300 es
+        #define FRAG_COLOR 0
+
+        precision highp float;
+        precision highp int;
+
+        uniform sampler2D diffuse;
+
+        in vec2 st;
+
+        layout(location = FRAG_COLOR) out vec4 color;
+
+        void main()
+        {
+            color = texture(diffuse, st);
+        }
+    </script>
+
+    <script>
+    (function()  {
+        'use strict';
+
+        var canvas = document.createElement('canvas');
+        canvas.width = Math.min(window.innerWidth, window.innerHeight);
+        canvas.height = canvas.width;
+        document.body.appendChild(canvas);
+
+        var gl = canvas.getContext( 'webgl2', { antialias: false } );
+        if (!gl) {
+            gl = canvas.getContext( 'experimental-webgl2', { antialias: false } );
+        }
+        var isWebGL2 = !!gl;
+        if(!isWebGL2)
+        {
+            document.getElementById('info').innerHTML = 'WebGL 2 is not available.  See <a href="https://www.khronos.org/webgl/wiki/Getting_a_WebGL_Implementation">How to get a WebGL 2 implementation</a>';
+            return;
+        }
+
+        // -- Divide viewport
+
+        var windowSize = {
+            x: canvas.width,
+            y: canvas.height
+        };
+
+        var CORNERS = {
+            TOP_LEFT: 0,
+            TOP_RIGHT: 1,
+            BOTTOM_RIGHT: 2,
+            BOTTOM_LEFT: 3,
+            MAX: 4
+        };
+
+        var Viewport = new Array(CORNERS.MAX);
+
+        Viewport[CORNERS.TOP_LEFT] = {
+            x: 0,
+            y: 0,
+            z: windowSize.x / 2,
+            w: windowSize.y / 2
+        };
+
+        Viewport[CORNERS.TOP_RIGHT] = {
+            x: windowSize.x / 2,
+            y: 0,
+            z: windowSize.x / 2,
+            w: windowSize.y / 2
+        };
+
+        Viewport[CORNERS.BOTTOM_RIGHT] = {
+            x: windowSize.x / 2,
+            y: windowSize.y / 2,
+            z: windowSize.x / 2,
+            w: windowSize.y / 2
+        };
+
+        Viewport[CORNERS.BOTTOM_LEFT] = {
+            x: 0,
+            y: windowSize.y / 2,
+            z: windowSize.x / 2,
+            w: windowSize.y / 2
+        };
+
+        // -- Initialize program
+
+        var program = createProgram(gl, getShaderSource('vs'), getShaderSource('fs'));
+
+        var uniformMvpLocation = gl.getUniformLocation(program,"mvp");
+        var uniformDiffuseLocation = gl.getUniformLocation(program,"diffuse");
+
+        // -- Initialize buffer
+
+        var positions = new Float32Array([
```

Comment:
I renamed this to `positions` because it is more precise than `vertices`.  In this sample, each vertex has both a position and texture coordinates.